src/Microsoft.AspNetCore.Rewrite/Internal/ConditionEvaluator.cs

```diff
@@ -24,18 +24,26 @@ public static MatchResults Evaluate(IEnumerable<Condition> conditions, RewriteCo
                     continue;
                 }
 
-                prevCond = condition.Evaluate(context, ruleMatch, prevCond);
+                var condResult = condition.Evaluate(context, ruleMatch, prevResult);
 
                 if (condition.OrNext)
                 {
-                    orSucceeded = prevCond.Success;
+                    orSucceeded = condResult.Success;
                 }
-                else if (!prevCond.Success)
+                else if (!condResult.Success)
                 {
-                    return prevCond;
+                    return condResult;
                 }
+
+                if (condResult.Success && trackAllCaptures && prevResult?.BackReferences != null)
+                {
+                    prevResult.BackReferences.Add(condResult.BackReferences);
+                    condResult.BackReferences = prevResult.BackReferences;
```

Comment:
Still need to re-do this logic so that we track the backreference collections directly rather than updating MatchResults.